pkg/command/autoscaling/list_test.go

```diff
@@ -0,0 +1,130 @@
+// Copyright © 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package autoscaling
+
+import (
+	"sort"
+	"strings"
+	"testing"
+	"time"
+
+	"gotest.tools/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+	"knative.dev/client/pkg/util"
+	"knative.dev/kn-plugin-admin/pkg"
+	"knative.dev/kn-plugin-admin/pkg/testutil"
+	"knative.dev/serving/pkg/autoscaler/config"
+)
+
+func checkListOutput(t *testing.T, data map[string]string, output string, noHeaders bool) {
+	config, err := config.NewConfigFromMap(data)
+	assert.NilError(t, err)
+
+	lines := strings.Split(output, "\n")
+	if !noHeaders {
+		assert.Check(t, util.ContainsAll(lines[0], "NAME", "VALUE"))
+	}
+
+	names := make([]string, 0, len(configNameValueOfMap))
+	for key := range configNameValueOfMap {
+		names = append(names, key)
+	}
+	sort.Strings(names)
+
+	start := 1
+	if noHeaders {
+		start = 0
+	}
+	for i, name := range names {
+		value := configNameValueOfMap[name](config)
+		assert.Check(t, util.ContainsAll(lines[i+start], name, value))
+	}
+}
+
+func TestDescribesDuration(t *testing.T) {
+	t0 := 123 * time.Second
+	assert.Equal(t, t0.String(), describeDuration(t0))
+
+	t1 := 60 * time.Second
+	assert.Equal(t, "1m", describeDuration(t1))
+
+	t2 := 3600 * time.Second
+	assert.Equal(t, "1h", describeDuration(t2))
+
+	t3 := 1238548 * time.Second
+	assert.Equal(t, t3.String(), describeDuration(t3))
+}
+
+func TestAutoscalingListDefaultValues(t *testing.T) {
```

Comment:
What if the value is not in config map? I don’t seem to see a test for that case?

Reply:
I use `NewConfigFromMap` function in my codes to create autoscaler config with the default values, the function is defined in **knative.serving** side, so it is not easy to mock config with a new field and value. But I improved `checkListOutput` function in my UT codes to make sure the output and config map have the same field amount, if the autoscaler config structure in knative.serving is changed, for example: adding or removing fields, our UT will be failed and then we can improve our config map accordingly.